src/globus_cli/commands/timer/create.py

```diff
@@ -0,0 +1,288 @@
+    type=str,
+    help=(
+        "An optional label for the transfer operation, up to 128 characters long. Must"
+        " contain only letters/numbers/spaces, and the following characters: - _ ,"
+    ),
+)
+@click.option(
+    "--stop-after-date",
+    required=False,
+    type=click.DateTime(formats=DATETIME_FORMATS),
+    help=("Stop running the transfer after this date"),
+)
+@click.option(
+    "--stop-after-runs",
+    required=False,
+    type=int,
+    help=("Stop running the transfer after this number of runs have happened"),
+)
+@click.option(
+    "--sync-level",
+    required=False,
+    type=int,
+    help=(
+        "Specify that only new or modified files should be transferred. The behavior"
+        " depends on the value of this parameter, which must be a value 0–3, as"
+        " defined in the transfer API: 0. Copy files that do not exist at the"
+        " destination. 1. Copy files if the size of the destination does not match the"
+        " size of the source. 2. Copy files if the timestamp of the destination is"
+        " older than the timestamp of the source. 3. Copy files if checksums of the"
+        " source and destination do not match."
+    ),
+)
+@click.option(
+    "--encrypt-data",
+    is_flag=True,
+    default=False,
+    help="Whether Transfer should encrypt data sent through the network using TLS",
+)
+@click.option(
+    "--verify-checksum",
+    is_flag=True,
+    default=False,
+    help=(
+        "Whether Transfer should verify file checksums and retry if the source and"
+        " destination don't match"
+    ),
+)
+@click.option(
+    "--preserve-timestamp",
+    is_flag=True,
+    default=False,
+    help=(
+        "Whether Transfer should set file timestamps on the destination to match the"
+        " origin"
+    ),
+)
+@click.option(
+    "--item",
+    "-i",
+    required=False,
+    type=(str, str, bool),
+    multiple=True,
+    help=(
+        "Used to specify the transfer items; provide as many of this option as files"
+        " to transfer. The format for this option is `--item SRC DST RECURSIVE`, where"
+        " RECURSIVE specifies, if this item is a directory, to transfer the entire"
+        " directory. For example: `--item ~/file1.txt ~/new_file1.txt false`"
+    ),
+)
+@click.option(
+    "--skip-source-errors",
+    is_flag=True,
+    default=False,
+    help=(
+        "Skip files or directories on the source endpoint that hit PERMISSION_DENIED or"
+        " FILE_NOT_FOUND errors."
+    ),
+)
+@click.option(
+    "--fail-on-quota-errors",
+    is_flag=True,
+    default=False,
+    help="Cancel the submitted transfer tasks if QUOTA_EXCEEDED errors are hit.",
+)
+@click.option(
+    "--recursive-symlinks",
+    required=False,
+    type=click.Choice(["ignore", "keep", "copy"], case_sensitive=False),
+    default="ignore",
+)
+@click.option(
+    "--items-file",
+    required=False,
+    type=str,
+    help="file containing table of items to transfer",
+)
```

Comment:
If we accept my argument above about `--item`, and internal consistency of the CLI, then this should become `--batch`.

Reply:
Yeah, I think I'm in agreement with both of these? I think this slightly predated the CLI feature of reading from files here so to accomplish the csv-file interface it looked different originally. The only thing is that the timer interface is unable to care about more than the `recursive` setting. Do you think it would make sense to still allow writing just a boolean in the file instead of the flag?

Reply:
> The only thing is that the timer interface is unable to care about more than the `recursive` setting. Do you think it would make sense to still allow writing just a boolean in the file instead of the flag?

I'm not sure I understand fully. Let's suppose I have a TransferData built with three items:
```
tdata = TransferData(...)
tdata.add_item("foo", "foo", recursive=True)
tdata.add_item("bar", "bar", recursive=True)
tdata.add_item("baz", "baz", recursive=False)  # default, but explicit for this example
```

Does the Timer service not support this mode of use? That seems unlikely to me based on the CSV reader, but maybe I'm mistaken.

If your concern is about `--recursive` vs `--batch` and how those behave, there's mutex behavior in `globus transfer` for `--recursive` and `--batch` with a specialized error message:
https://github.com/globus/globus-cli/blob/b16ef4c18fcfbfa1c76136209ba76b3223523480/src/globus_cli/commands/transfer.py#L321-L326

I think the mutex_option_group decorator could support this in the future, if we give it the ability to customize the error message used.

Reply:
No it does, I'm trying to say that (due to the AP interface) Timer at most can use those three values for each line (path, path, recursive)—any other flag should apply to all of them. So `--recursive` is the only flag that should be allowed in the file. I guess it makes sense, I just find the csv-ish interface more intuitive than putting the flags in the file.

Reply:
I don't mean to defend the "inner command" method of parsing as perfect. It's odd at best.

My worry with CSV data is that it won't be easy to use when there are many options, mostly used infrequently. We may one day see data like
```
foo,bar
bar,baz,,,auto,,true,,md5
```
because it's a format that may accept the optionality of the rightmost columns, but doesn't have a per-row way of mapping keys to values.

Obviously, that example is extreme, but I like situations in which order doesn't matter. `--recursive a b` and `a b --recursive` are the same.

Reply:
I agree with your example, and I'm not saying we should look to apply this in general. Specifically for the case with timer where you have exactly 3 values I would lean slightly towards continuing to support the CSV mode there. Do you feel like if the CSV behavior exists for Timer but not elsewhere that it would be prohibitively weird?

Reply:
Let me separate this into two separate points, because I think your last message is very-much on-target.

> Specifically for the case with timer where you have exactly 3 values I would lean slightly towards continuing to support the CSV mode there.

Yes, and I agree that with exactly 3 known columns, CSV is a nice format. However...

> Do you feel like if the CSV behavior exists for Timer but not elsewhere that it would be prohibitively weird?

Maybe sub "weird" for "creates surprising results", but yes. 😄 

Let's suppose that `globus transfer $src $dest --batch foo.txt` works today.
I'd like for `globus timer create transfer $src $dest --batch foo.txt --interval 10m ...` to also work, and vice versa. [1]

That means that because `globus transfer --batch` has its existing line-parsing behavior, we should support it in `globus timer create transfer`.

The "out" for this, if we much prefer the CSV usage, is to support both. And if we're going to do the work for that, I think we need `globus transfer` to support the same. If we implement both parsing styles, we need an option or usage to set the behavior. We could have `--batchmode [auto|txt|csv]` where `auto` has fancy behavior.

But all in all, this seems like it's going down a path of a new and separate feature to support CSV as a format for transfer batches. I'm not sure I'm convinced that the (IMO, mild) niceness of CSV for the simple case is worth it vs the cost of supporting it in both commands?

----

[1] I'm aware that this could be messed up by `--external-checksum`, but IMO that's a reason to push adding that to the AP and making it work, not to change how the CLI tries to make these commands appear related.

Reply:
> I'd like for globus timer create transfer $src $dest --batch foo.txt --interval 10m ... to also work

This is kind of what I'm getting at, that for this `foo.txt` is limited to the src, dst, recursive triplet. This would kind of be similar to [1], that the AP should just do it the, really, _better_ way. But as it is the batch behavior is distinct between the two. So I'm trying to get to what seems the least confusing given that present limitation.

Reply:
What I would consider ideal is that -- and let's set aside the details of how the parsing happens -- both `globus transfer` and `globus timer create transfer` use the same parsing methodology and same parse function, potentially parametrized for their different cases.

I'd like to see the `globus transfer` case implemented as:
```python
if batch:
    tdata = magical_parse_batch_input(batch)
else: ...
```
and the `globus timer create transfer` case implemented as:
```python
if batch:
    tdata = magical_parse_batch_input(batch, external_checksum_supported=False)
else:
    ...
```

That way, we don't have two different overall parse behaviors, and it's possible to formulate a single input which is valid for both. Even if we have an unfortunate situation like `--external-checksum` in which some inputs are valid for one but not the other, it makes it possible for this to work some of the time.

In principle, I would be fine with `magical_parse_batch_input` doing CSV, shlexed line-reads, newline-delimited JSON, any other "reasonable" format, or even detection based on the first line of input (imagine trying different parses until one works). In practice, I don't know that we want to bother about having multiple ways of parsing those inputs today.